server/models/Transaction.js

```diff
@@ -47,7 +47,14 @@ export default (Sequelize, DataTypes) => {
         },
         onDelete: 'SET NULL',
         onUpdate: 'CASCADE',
-        allowNull: false,
+        allowNull: true, // we allow CreatedByUserId to be null but only on refund transactions
+        validate: {
+          isValid(value) {
```

Comment:
I was trying to find the documentation for `isValid` and was unable to find it. I just wanted to check if it was the good practice to throw an exception there.
 
Let me know if you have any link. Otherwise, looks good.

Reply:
Not really a Sequelize method but instead a customer validator (bottom of the table): https://sequelize.org/master/manual/validations-and-constraints.html#validators